src/main/java/kmql/QueryCompleter.java

```diff
@@ -0,0 +1,61 @@
+package kmql;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.UncheckedIOException;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map.Entry;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.jline.reader.Candidate;
+import org.jline.reader.Completer;
+import org.jline.reader.LineReader;
+import org.jline.reader.ParsedLine;
+import org.jline.reader.impl.completer.StringsCompleter;
+
+/**
+ * A {@link Completer} implementation for interactive kmql session.
+ */
+public class QueryCompleter implements Completer {
+    private final StringsCompleter delegate;
+
+    public static QueryCompleter from(Engine engine) {
```

Comment:
How about rather take a `Database` with adding it a method to obtain the containing tables so this completer will be consistent with the registry we gave to construct `Database`?

Reply:
Sounds good but then how should we pass Database to QueryCompleter ?
Currently I instantiate QueryCompleter inside `Kmql#call` but `Database` is not visible there.

Is that acceptable to expose `Engine#db` ?

Reply:
> Is that acceptable to expose `Engine#db`

Already exposed I think?

Reply:
Oh overlooked `@Getter`... I see.